src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs

```diff
@@ -606,5 +645,85 @@ private static byte[] GetUtf8BytesFromString(string s)
 
             return bytes;
         }
+
+        public Utf8String TrimStart()
+        {
+            CodePointEnumerator it = GetCodePointEnumerator();
+            while (it.MoveNext() && UnicodeCodePoint.IsWhitespace(it.Current))
+            {
+            }
+
+            return Substring(it.PositionInCodeUnits);
+        }
+
+        public Utf8String TrimStart(UnicodeCodePoint[] trimCodePoints)
```

Comment:
We need to get with the right design early to detect flaws early. I added this to match string and so that API already exists and you can start using the API even though it is not implemented yet (at least you know what to expect in the future :) ).